src/features/DisplayControls/Vitals/components/Vitals.jsx

```diff
@@ -0,0 +1,291 @@
+    spO2: <FormattedMessage id={"SPO2"} defaultMessage={"SpO2"} />,
+    BMI: <FormattedMessage id={"BMI"} defaultMessage={"BMI"} />,
+  };
+
+  var latestDate = null;
+
+  const setDateAndTime = (latestDateAndTime) => {
+    const dateAndTime = formatDate(latestDateAndTime).split(" ");
+    setVitalsDate(
+      formatDateAsString(new Date(latestDateAndTime), "DD/MM/YYYY")
+    );
+    setVitalsTime(dateAndTime.slice(3).join(" ").toUpperCase());
+  };
+
+  function getLatestDate(tabularData) {
+    for (const dateKey in tabularData) {
+      if (latestDate === null || dateKey > latestDate) {
+        latestDate = dateKey;
+      }
+    }
+  }
+
+  const mapVitalsData = (VitalsList) => {
+    var mappedVitals = {};
+    if (VitalsList.tabularData) {
+      const VitalsValues = VitalsList.tabularData;
+      getLatestDate(VitalsValues);
+      if (latestDate !== null) {
+        setDateAndTime(latestDate);
+        mappedVitals = {
+          Temp: {
+            value: VitalsValues[latestDate].Temperature?.value,
+            abnormal: VitalsValues[latestDate].Temperature?.abnormal,
+          },
+          HeartRate: {
+            value: parseInt(VitalsValues[latestDate].Pulse?.value, 10),
+            abnormal: VitalsValues[latestDate].Pulse?.abnormal,
+          },
+          SystolicPressure: {
+            value: parseInt(
+              VitalsValues[latestDate]["Systolic Blood Pressure"]?.value,
+              10
+            ),
+            abnormal:
+              VitalsValues[latestDate]["Systolic Blood Pressure"]?.abnormal,
+          },
+          DiastolicPressure: {
+            value: parseInt(
+              VitalsValues[latestDate]["Diastolic Blood Pressure"]?.value,
+              10
+            ),
+            abnormal:
+              VitalsValues[latestDate]["Diastolic Blood Pressure"]?.abnormal,
+          },
+          Height: {
+            value: parseInt(VitalsValues[latestDate].HEIGHT?.value, 10),
+            abnormal: VitalsValues[latestDate].HEIGHT?.abnormal,
+          },
+          Weight: {
+            value: parseInt(VitalsValues[latestDate].WEIGHT?.value, 10),
+            abnormal: VitalsValues[latestDate].WEIGHT?.abnormal,
+          },
+          RespiratoryRate: {
+            value: parseInt(
+              VitalsValues[latestDate]["Respiratory Rate"]?.value,
+              10
+            ),
+            abnormal: VitalsValues[latestDate]["Respiratory Rate"]?.abnormal,
+          },
+          SpO2: {
+            value: parseInt(VitalsValues[latestDate].SpO2?.value, 10),
+            abnormal: VitalsValues[latestDate].SpO2?.abnormal,
+          },
+          BMI: {
+            value: VitalsValues[latestDate].BMI?.value,
+            abnormal: VitalsValues[latestDate].BMI?.abnormal,
+          },
+        };
+      }
+    }
+    return mappedVitals;
+  };
+
+  const handleVitalUnits = (units) => {
+    units.forEach((unit) => {
+      setVitalUnits((oldUnits) => {
```

Comment:
instead of updating vital units inside "for loop", can we update the final object in state after the loop